### PR TITLE
Add optional reason when using the stopgame command.

### PR DESCRIPTION
**Pull request by Pentarctagon**
So now: `/query stopgame <nick> [<reason>]`. The reason is displayed to the players of the game on the savegame prompt dialog.

### CODE REVIEW

File: src/playturn.cpp [hunk 1]
```diff
@@ -156,7 +156,8 @@ turn_info::PROCESS_DATA_RESULT turn_info::process_network_data(const config& cfg
 		game_display::get_singleton()->get_chat_manager().remove_observer(observer_quit["name"]);
 	}
 	else if (cfg.child("leave_game")) {
-		throw ingame_wesnothd_error("");
+		const bool has_reason = cfg.child("leave_game").has_attribute("reason");
+		throw ingame_wesnothd_error(has_reason ? cfg.child("leave_game").get("reason")->str() : "");
```
Comment (gfgtdf) [inline src/playturn.cpp]: even if you check for has_attribute before, is still better to use ["reason"] over get("reason")

Comment (Pentarctagon) [reply src/playturn.cpp]: Done.

File: src/server/server.cpp [hunk 2]
```diff
@@ -2869,10 +2871,13 @@ void server::delete_game(int gameid)
 	}
 
 	// send users in the game a notification to leave the game since it has ended
-	static simple_wml::document leave_game_doc("[leave_game]\n[/leave_game]\n", simple_wml::INIT_COMPRESSED);
+	simple_wml::document leave_game_doc_reason("[leave_game]\n[/leave_game]\n", simple_wml::INIT_COMPRESSED);
```
Comment (soliton-) [inline src/server/server.cpp]: Not sure if it makes much of a difference but I would keep the static variable for the common "no reason" case and use another document (without `INIT_COMPRESSED`) for the case with a dynamic reason.

Comment (Pentarctagon) [reply src/server/server.cpp]: Creating the document as:
`simple_wml::document leave_game_doc_reason(simple_wml::string_span("[leave_game]\n[/leave_game]\n"));`
results in the server dying with:
`error server: terminated by C++ exception: failed to uncompress`
when I provide a reason to the stopgame command.

Comment (soliton-) [reply src/server/server.cpp]: That calls a different constructor. I meant using INIT_STATIC instead.

Comment (Pentarctagon) [reply src/server/server.cpp]: Ah, alright, updated.

**Comment by soliton-** [comment]
Right, the WML is of course generated on the client and the server just copies it. Looks all good to me then.

**Comment by soliton-** [comment]
Feel free to backport to 1.14.

**Pull request merged**